src/utilities/tests/test_html.py

```diff
@@ -23,3 +24,199 @@ def test_textify_donot_ignore_links(self):
         html = """<p>Hello, <a href='https://www.google.com/earth/'>world</a>!</p>"""
         output = textify(html)
         self.assertEqual(output, """Hello, [world](https://www.google.com/earth/)!\n\n""")
+
+
+class LinkTextExtractor(HTMLParser):
+    def __init__(self):
+        super().__init__()
+        self.link_texts = []
+
+    def handle_data(self, data):
+        self.link_texts.append(data)
+
+    def get_text(self):
+        return "".join(self.link_texts)
+
+
+class UrlizeTests(SimpleTestCase):
+    def test_basic(self):
+        """
+        Test that a simple URL is correctly converted into an anchor tag
+        and the URL text is preserved.
+        """
+        text = "Visit www.example.com"
+        result = urlize(text)
+        self.assertIn('<a href="http://www.example.com"', result)
+        self.assertIn('www.example.com', result)
+
+        text = "1.www.example.com"
+        result = urlize(text)
+        self.assertIn('<a href="http://www.example.com"', result)
```

Comment:
All these `assertIn` seem redundant with the assertTrue after it, since if the second one is true, obviously it contains the substring tested the line before?